src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs

```diff
@@ -0,0 +1,235 @@
+            { 
+                if (value is OpenApiArray array)
+                {
+                    var list = new List<string>(array.Capacity);
+
+                    foreach (var item in array)
+                    {
+                        if (item is OpenApiObject obj)
+                        {
+                            // has keys, "value", and "displayName"
+                            if (obj.TryGetValue("value", out var value2))
+                            {
+                                if (value2 is OpenApiString str)
+                                {
+                                    list.Add(str.Value);
+                                    continue;
+                                }
+                            }
+                        }
+
+                        throw new NotImplementedException($"Unrecognized x-ms-enum-values schema");
+                    }
+
+                    return list.ToArray();
+                }
+            }
+
+            return null;
+        }
+
+        public static bool IsTrigger(this OpenApiOperation op)
+        {
+            var isTrigger = op.Extensions.ContainsKey("x-ms-trigger");
+            return isTrigger;
+        }
+
+        public static bool TryGetDefaultValue(this OpenApiParameter param, out string defaultValue)
+        {
+            var x = param.Schema.Default;
+            if (x == null)
+            {
+                defaultValue = null;
+                return false;
+            }
+
+            if (x is OpenApiString str)
+            {
+                defaultValue = str.Value;
+                return true;
+            }
+
+            if (x is OpenApiInteger intVal)
+            {
+                defaultValue = intVal.Value.ToString();
+                return true;
+            }
+
+            if (x is OpenApiDouble dbl)
+            {
+                defaultValue = dbl.Value.ToString();
+                return true;
+            }
+
+            if (x is OpenApiBoolean b)
+            {
+                defaultValue = b.Value.ToString();
+                return true;
+            }
+
+            throw new NotImplementedException($"Unknown default value type {x.GetType().FullName}");
+        }
+
+        public static bool HasDefaultValue(this OpenApiParameter param)
+        {
+            return param.Schema.Default != null;
+        }
+
+        // Internal parameters are not showen to the user. 
+        // They can have a default value or be special cased by the infrastructure (like "connectionId").
+        public static bool IsInternal(this OpenApiParameter param) => param.Extensions.TryGetValue("x-ms-visibility", out _);
+
+        // See https://swagger.io/docs/specification/data-models/data-types/
+        public static FormulaType ToFormulaType(this OpenApiSchema schema)
+        {
+            switch (schema.Type)
+            {
+                case "string": return FormulaType.String;
```

Comment:
yes - although any missing case will get the not-implemented on line 172 below.  So as we add test cases that excercise , we'll fill these in.

Reply:
This is a bit tricky from a back compat perspective, however. They'll currently parse as Strings, but later will change to DateTime. We're preview, so that's ok, but we should let consumers know about that.